crates/sui-types/src/ed25519.rs

```diff
@@ -0,0 +1,608 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
```

Comment:
Separated Ed25519 and BLS12381 functionalities into separate files to allow switching between the two in the future